### PR TITLE
Langauge Server Support

### PR DESCRIPTION
**Pull request by gdotdesign**
This PR adds support for the language server with minimal configuration for testing purposes.

I would like to release 0.4.0 with this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,6 +17,9 @@
     "onCommand:mint.init"
   ],
   "main": "./out/src/extension",
+  "dependencies": {
+    "vscode-languageclient": "^6.1.3"
```
Comment (gdotdesign) [inline package.json:21]: I don't know if it's needed or not, it was needed for testing the extension.

Comment (Sija) [reply package.json:21]: Maybe let's move to the `devDependencies` then?

File: src/extension.ts [hunk 2]
```diff
@@ -26,9 +31,36 @@ export async function activate(
   vscode.commands.registerCommand("mint.start", cmd.mintStartCommand);
   vscode.commands.registerCommand("mint.test", cmd.mintTestCommand);
   vscode.commands.registerCommand("mint.version", cmd.mintVersionCommand);
+
+  let binaryLocation : string = vscode.workspace.getConfiguration('mint_language_server').get('location')
```
Comment (Sija) [inline src/extension.ts:35]:
```suggestion
  const binaryLocation : string = vscode.workspace.getConfiguration('mint_language_server').get('location')
```

File: src/extension.ts [hunk 3]
```diff
@@ -26,9 +31,36 @@ export async function activate(
...
+    } else {
+      vscode.window.showErrorMessage('Mint binary not found! You specified ' + binaryLocation);
+    }
+  }
 }
 
 export async function deactivate(isRestart: boolean = false): Promise<void> {
   /// Set context deactivated
   vscode.commands.executeCommand("setContext", "mint:isActivated", false);
+  if (client) { client.stop() }
```
Comment (Sija) [inline src/extension.ts:65]:
```suggestion
  if (client) {
    client.stop()
  }
```

File: package.json (hunk 1 above)
Comment (s0kil) [reply package.json:21]: Looks good.

File: package.json [hunk 4]
```diff
@@ -35,6 +38,16 @@
     "Programming Languages"
   ],
   "contributes": {
+    "configuration": {
+      "type": "object",
+      "title": "Mint Language Server",
+      "properties": {
+        "mint_language_server.location": {
```
Comment (s0kil) [inline package.json:45]: Prefer camelCase.

**Review by s0kil — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (s0kil) [reply package.json:45]: mint.languageServer.location

File: src/extension.ts [hunk 5]
```diff
@@ -26,9 +31,40 @@ export async function activate(
...
   vscode.commands.registerCommand("mint.test", cmd.mintTestCommand);
   vscode.commands.registerCommand("mint.version", cmd.mintVersionCommand);
+
+  const binaryLocation : string = vscode.workspace.getConfiguration('mint.languageServer').get('location')
+
+  if (binaryLocation) {
+    if (fs.existsSync(binaryLocation)) {
+      // Create the language client
+      client = new LanguageClient(
+        'mint_Language_Server',
```
Comment (Sija) [inline src/extension.ts:41]:
```suggestion
        'mint_language_server',
```

Comment (gdotdesign) [reply src/extension.ts:41]: :heavy_check_mark: 61ca6e626f13b34393431733e558e3c40c425ed1

**Pull request merged**